### ISSUE TITLE
Generate typescript app not starting

### ISSUE THREAD

**Issue by kamo93** [fastify/fastify-cli#513]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported

### Fastify version

4.0.0

### Plugin version

3.0.0

### Node.js version

14.17.0

### Operating system

Linux

### Operating system version (i.e. 20.04, 11.3, 10)

Ubuntu 20.04.4

### Description

When trying to start a just generate app using typescript the browser shows 
![image](https://user-images.githubusercontent.com/18643841/175788034-782f4dcb-f97e-4f7f-8e78-e01d4286e396.png)

When trying to generate and start a js fastify app everything work fine  but with typescript doesn't start.

This is what  it shows the console.
![image](https://user-images.githubusercontent.com/18643841/175788160-b27ab9f2-b672-437f-a58a-32de148c701f.png)

### Steps to Reproduce

Generate an app, and trying to start it.
```
fastify generate <app-name> --lang=ts

npm install

npm start dev
```

### Expected Behavior

Start the app normally

**Comment by kamo93** [comment]
The only work around that I found was to run `npm run watch:ts` and `npm run dev:start` on different terminals. But sometimes while the new dist/app.js gets build the server response unavailable service just for a moment.

**Comment by mcollina** [comment]
Where did you see you should run `npm start dev`?  Running `npm start` works just fine.

I can see the problem you are describing with `npm run dev`. Is this correct?

**Comment by mcollina** [comment]
@fastify/typescript could you take a look here? I don't understand why `npm run dev` is not really working.

**Comment by kamo93** [comment]
Sorry my mistake. The problem occurs with `npm run dev`

**Comment by BigGillyStyle** [comment]
FWIW...I also am seeing this same problem in this local config:

MacOS: Monterey 12.4
Node: 16.15.1
NPM: 8.13.1

```shell
$ npm ls
├── @fastify/autoload@5.0.0
├── @fastify/sensible@5.1.0
├── @types/node@16.11.41
├── @types/tap@15.0.7
├── concurrently@7.2.2
├── fastify-cli@4.2.0
├── fastify-plugin@3.0.1
├── fastify-tsconfig@1.0.1
├── fastify@4.1.0
├── tap@16.3.0
├── ts-node@10.8.1
└── typescript@4.7.4
```

**Comment by mcollina** [comment]
I have no clue why this is happening and not really much time to look into it right now. A PR that fixes this would be awesome.

**Comment by wlinna** [comment]
For me it works when I run `npm run dev` the first time, but after I modify files and save, and `tsc -w` compiles, the server doesn't seem to start anymore. I also sometimes see that Service Unavailable though.

For the few weeks that I've used  `fastify-cli ^3.1.0` and `fastify ^3.0.0` this never happened, but after upgrading  it stopped working.

**Comment by BigGillyStyle** [comment]
I have done some investigation locally, and `npm run dev` does not work for a TS-generated project with `fastify-cli` 4.2.0 but _does_ with 4.1.1 and earlier (went back incrementally to 3.1.0 and found this to be true in all cases).  In particular in 4.2.0 and [this line](https://github.com/fastify/fastify-cli/blob/master/lib/watch/index.js#L54):

```typescript
      if (childs.length === 0 && !allStop) {
        childs.push(run('restart'))
      }
```

...is in an endless loop of restarts.  The difference between 4.1.1 and 4.2.0 appears to be (based on console.log statements I inserted) due to `childs.length` being 0 in 4.2.0 and 1 in 4.1.1.  I'll see if I can do some more debugging.

**Comment by BigGillyStyle** [comment]
If I revert the one-line change in [this PR](https://github.com/fastify/fastify-cli/pull/511), then my local testing with all the other 4.2.0 changes shows that this issue is fixed.

**Comment by BigGillyStyle** [comment]
If it's desired, I can work on a PR to revert this (or whatever else might be recommended)

**Comment by BigGillyStyle** [comment]
> Would you like to send a Pull Request to address this issue? It would be amazing to have a unit test too!

Yes, I can do that.  I'd like to spend at least some time understanding the code paths that result in a `null` versus `0` code in order to hopefully prevent a regression of this nature in the future.  Some unit tests that made those distinctions clear would be (I think) ideal.

**Comment by fox1t** [comment]
I am not sure why that change was introduced in the first place :/

**Comment by mcollina** [comment]
Note that `!code !== 0` always evaluates to `true`, independently of the value of `code`. It's a bug.

**Comment by fox1t** [comment]
Yes, but did we understand how it affects the TS template restart? (I didn't :/)

**Comment by mcollina** [comment]
No clue on that yet.

**Comment by BigGillyStyle** [comment]
Actually...if you go back to the file history, the bug was that it always evaluated to `false`:

https://github.com/fastify/fastify-cli/blob/v3.1.0/lib/watch/index.js#L51
```javascript
      if (!code === 0) {
        stop()
      }
```

And indeed...in my own testing locally of both JS and TS generated projects using fastify-cli 4.2.0...if I just remove that code block altogether, everything works as expected (on initial start of server as well as after making a code change in the generated app)

**Comment by Vistey** [comment]
try this
```
# package.json
{
  "dependencies": {
    ...
    "fastify-cli": "4.1.0"
  },
}
```

**Comment by Uzlopak** [comment]
Well in hindsight I should have removed the code and avoided so the regression.
